test/test.hs

```diff
@@ -895,6 +895,68 @@ tests = testGroup "hevm"
           propagated = Expr.constPropagate t
         assertEqualM "Must contain PBool False" True ((PBool False) `elem` propagated)
   ]
+  , testGroup "inequality-propagation-tests" [
+      test "PLT-detects-impossible-constraint" $ do
+        let
+          -- x < 0 is impossible for unsigned integers
+          t = [PLT (Var "x") (Lit 0)]
+          propagated = Expr.constPropagate t
+        assertEqualM "Must contain PBool False" True ((PBool False) `elem` propagated)
+    , test "PLT-overflow-check" $ do
+        let
+          -- maxLit < y is impossible
+          t = [PLT (Lit 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff) (Var "y")]
+          propagated = Expr.constPropagate t
+        assertEqualM "Must contain PBool False" True ((PBool False) `elem` propagated)
+    , test "PGT-detects-impossible-constraint" $ do
+        let
+          -- x > maxLit is impossible
+          t = [PGT (Var "x") (Lit 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff)]
+          propagated = Expr.constPropagate t
+        assertEqualM "Must contain PBool False" True ((PBool False) `elem` propagated)
+    , test "PGT-overflow-check" $ do
+        let
+          -- 0 > y is impossible
+          t = [PGT (Lit 0) (Var "y")]
+          propagated = Expr.constPropagate t
+        assertEqualM "Must contain PBool False" True ((PBool False) `elem` propagated)
```

Comment:
I am not convinced we need to detect these in `constPropagate`.
I think these should be simplified to `False` in earlier stages, in prop simplification.